api/callbacks.go

```diff
@@ -365,25 +365,31 @@ var querier_vtable = C.Querier_vtable{
 
 // contract: original pointer/struct referenced must live longer than C.GoQuerier struct
 // since this is only used internally, we can verify the code that this is the case
-func buildQuerier(q Querier) C.GoQuerier {
+func buildQuerier(q Querier, gm GasMeter) C.GoQuerier {
 	return C.GoQuerier{
+		gas_meter: (*C.gas_meter_t)(unsafe.Pointer(&gm)),
 		state:  (*C.querier_t)(unsafe.Pointer(&q)),
 		vtable: querier_vtable,
 	}
 }
 
 //export cQueryExternal
-func cQueryExternal(ptr *C.querier_t, request C.Buffer, result *C.Buffer) (ret C.GoResult) {
+func cQueryExternal(ptr *C.querier_t, gasMeter *C.gas_meter_t, usedGas *C.uint64_t, request C.Buffer, result *C.Buffer) (ret C.GoResult) {
 	defer recoverPanic(&ret)
-	if result == nil {
+	if ptr == nil || gasMeter == nil || usedGas == nil || result == nil {
 		// we received an invalid pointer
 		return C.GoResult_BadArgument
 	}
 
 	// query the data
+	gm := *(*GasMeter)(unsafe.Pointer(gasMeter))
```

Comment:
If we want to make this minimal and tailored to the cosmos-sdk (which I think we agree on), we can just add a method to `Querier` to check `GasConsumed()`.

Reply:
This is the Querier: https://github.com/CosmWasm/wasmd/blob/master/x/wasm/internal/keeper/query_plugins.go#L12-L15

Reply:
That we pass into go-cosmwasm: https://github.com/CosmWasm/wasmd/blob/master/x/wasm/internal/keeper/keeper.go#L134-L141

Reply:
Ctx is a public member and we can just extend the interface to expose one method

Reply:
ahhh, cool, i'll play around with that :)

Reply:
I was going to expose the entire `GasMeter` object from the `Querier` interface, but for this to work, we need to redefine `Querier` in `types/queries.go` as:

```go
type Querier interface {
	Query(request QueryRequest) ([]byte, error)
	GasMeter() api.GasMeter
}
```

But referring to `api.GasMeter` from that context creates a cyclical reference, which Go doesn't allow :/ In order to refer to `GasMeter` here, we would need to redefine it (or pull in the cosmos sdk as a dependency).

For now, i'll just expose the `GasUsed` method.